end2end-test-examples/gcs/src/main/java/io/grpc/gcs/JavaApiServicesClient.java

```diff
@@ -135,17 +137,15 @@ public void makeRandomMediaRequest(ResultTable results, int threadId) throws IOE
         req.setReturnRawInputStream(false);
         MediaHttpDownloader mediaHttpDownloader = req.getMediaHttpDownloader();
         mediaHttpDownloader.setDirectDownloadEnabled(true);
-        ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream();
         req.executeMedia().download(byteArrayOutputStream);
-        buff.put(byteArrayOutputStream.toByteArray());
       } catch (IOException ioException) {
         System.err.println(ioException);
       }
       long dur = System.currentTimeMillis() - start;
-      if (buff.remaining() > 0) {
-        logger.warning("Got remaining bytes: " + buff.remaining());
+      if (byteArrayOutputStream.toByteArray().length > 0) {
+        logger.warning("Got remaining bytes: " + byteArrayOutputStream.toByteArray().length);
```

Comment:
I don't think this is checking what you actually want to check.

I think you want to check if the actual result is the expected size.

```java
int expectedSize = args.buffSize * 1024;
ByteArrayOutputStream baos = new ByteArrayOutputStream();
// ... run stuff
byte[] actual = baos.toByteArray();
int actualSize = actual.length;

if (actualSize != expectedSize) {
  logger.warning(String.format("Actual bytes size differed from expected. expected: %d actual: %d", expectedSize, actualSize));
}
```

Reply:
Addressed. thanks Ben i def misunderstood what this condition was doing.